browser-extension/common/src/custom-elements/test-rule-widget/implicit-test-rule-widget/index.ts

```diff
@@ -36,6 +36,7 @@ class RQImplicitTestRuleWidget extends RQTestRuleWidget {
 
   addWidgetListeners() {
     this.addEventListener("new-rule-applied", (evt: CustomEvent) => {
+      this.setAttribute("applied_listener_active", "true");
```

Comment:
We should not use attribute for this.We should setup custom events. When the listeners are active, only then notify the widget with all the applied rules using the notify function. That way we wont have 2 sources for the applied rules